gstd-meta/tests/json.rs

```diff
@@ -0,0 +1,378 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![no_std]
+
+// May be replaced with `use gstd_meta::*;`
+use gstd_meta::{to_json, types, TypeInfo};
+
+extern crate alloc;
+use alloc::{string::String, vec::Vec};
+
+#[allow(unused)]
+#[derive(TypeInfo)]
+struct SaltAmount {
+    value: u8,
+}
+
+#[allow(unused)]
+#[derive(TypeInfo)]
+struct Meat {
+    name: String,
+    salt: Option<SaltAmount>,
+}
+
+#[allow(unused)]
+#[derive(TypeInfo)]
+struct Egg {
+    weight: u32,
+    ostrich: bool,
+}
+
+#[allow(unused)]
+#[derive(TypeInfo)]
+struct Sauce {
+    eggs: Vec<Egg>,
+    salty: Result<SaltAmount, SaltAmount>,
+}
+
+#[allow(unused)]
+#[derive(TypeInfo)]
+struct Meal {
+    steak: Meat,
+    mayonnaise: Sauce,
+}
+
+// Function for more visual testing
+fn check(types: Vec<gstd_meta::prelude::MetaType>, expectation: &'static str) {
+    assert_eq!(
+        to_json(types),
```

Comment:
Just as a reminder for the future. The idiomatic way to call function is `module::function()`.
https://doc.rust-lang.org/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html